llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -963,6 +964,39 @@ class YkIRWriter {
     InstIdx++;
   }
 
+  /// Serialise a cast-like instruction.
+  void serialiseTruncInst(TruncInst *I, FuncLowerCtxt &FLCtxt, unsigned BBIdx,
+                         unsigned &InstIdx) {
+    // We don't support vector icmp.
+    if (I->getOperand(0)->getType()->isVectorTy()) {
+      serialiseUnimplementedInstruction(I, FLCtxt, BBIdx, InstIdx);
+      return;
+    }
+
+    DataLayout DL(&M);
+    TypeSize SrcSize = DL.getTypeSizeInBits(I->getSrcTy());
+    TypeSize DstSize = DL.getTypeSizeInBits(I->getDestTy());
+
+    if (SrcSize <= DstSize) {
```

Comment:
I think this can be an assert, since it's actually invalid LLVM IR if this isn't the case.

Reply:
Yeah, I wasn't super sure this was the case. Will fix.